plugins/lua/zone.lua

```diff
@@ -1216,10 +1216,188 @@ function RetireLocationOverlay:confirm_retire()
         curry(retire, location))
 end
 
+-- -------------------
+-- AnimalActionsWidget
+--
+
+local YES = 'Yes'
+local NO = 'No'
+
+-- Make sure an animal unit of your civ is selected
+local function check_valid_unit()
+    local unit = dfhack.gui.getSelectedUnit(true)
+    return unit
+        and dfhack.units.isFortControlled(unit)
+        and dfhack.units.isAlive(unit)
+        and dfhack.units.isAnimal(unit)
+end
+
+local function is_geldable()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- If the unit is not male or already gelded do nothing
+    return dfhack.units.isGeldable(unit)
+end
+
+local function is_not_pet()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    return not dfhack.units.isPet(unit)
+end
+
+local function is_avail_adoption()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- Make sure not already a pet and not a cat as they adopt owner
+    return not dfhack.units.isPet(unit) and unit.race ~= 171
+end
+
+AnimalActionsWidget=defclass(AnimalActionsWidget, overlay.OverlayWidget)
+AnimalActionsWidget.ATTRS {
+    desc = "Add options to tamed animals view sheet",
+    default_pos={x=-42,y=37},
+    default_enabled=true,
+    viewscreens='dwarfmode/ViewSheets/UNIT/Overview',
+    frame={w=23, h=6},
+}
+
+-- The above function already handles checking if valid unit
+-- so just set slaughter flag
+function AnimalActionsWidget:set_slaughter_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option then
+        unit.flags2.slaughter = true
+    else
+        unit.flags2.slaughter = false
+    end
+end
+
+-- set units marked for gelding flag
+function AnimalActionsWidget:set_geld_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option then
+        unit.flags3.marked_for_gelding = true
+    else
+        unit.flags3.marked_for_gelding = false
+    end
+end
+
+-- set available for adoption flag
+function AnimalActionsWidget:set_adoption_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option then
+        unit.flags3.available_for_adoption = true
+    else
+        unit.flags3.available_for_adoption = false
+    end
+end
+
+-- Check current flag status of animal to dynamically set on/off
+function AnimalActionsWidget:get_butcher(unit)
```

Comment:
These get functions are unused and can be removed

Reply:
Removed functions